File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.js
function solution(keyinput, board) {
    let arr = [0, 0];
  
    keyinput.forEach((x, i) => {
        switch(x) {
            case "up": arr[1]+1 <= ~~board[1]/2 ? arr[1] += 1 : arr[1];
                  break;
            case "down": arr[1]-1 >= -~~(board[1]/2) ? arr[1] -= 1 : arr[1];
                  break;
            case "left": arr[0]-1 >= -~~(board[0]/2) ? arr[0] -= 1 : arr[0];
                  break;
            case "right": arr[0]+1 <= ~~board[0]/2 ? arr[0] += 1 : arr[0];;
                  break;
        }
    })
    
    return arr;
}

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    move = {"up": [0, 1], "down": [0, -1], "left": [-1, 0], "right": [1, 0]}
    user = [0, 0]
    [board_x, board_y] = [board[0] // 2, board[1] // 2]

    for m in keyinput:
        if abs(user[0] + move[m][0]) <= board_x:
            user[0] += move[m][0]

        if abs(user[1] + move[m][1]) <= board_y:
            user[1] += move[m][1]

    return user